### PR TITLE
CI: Bump go version, linters and fix minor issues

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Bumps CI Go and golangci-lint versions, adds new linters and fixes minor issues. Some notes:

- nolintlint enforced directive: https://github.com/golangci/golangci-lint/issues/1658
- Prevents slowloris DoS vuln with ReadHeaderTimeout param
- Uses net package to split/join hosts for IPv6 support
- Fixes non-closed files on tests which polluted temp dir on systems
- Automated comments on exported funcs/methods and structs (same as golint ./...)
- Fixes Bitstamp / Bithumb / BTSE test issues
- gofmt doc issues

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -38,8 +38,9 @@ linters:
     - errchkjson
     - errname
 #   - errorlint
+    - execinquery
 #   - exhaustive
-#   - exhaustivestruct
+#   - exhaustivestruct # deprecated
```
Comment (gloriousCode) [inline .golangci.yml:43]:
```suggestion
#   - exhaustivestruct // deprecated
```
For consistency's sake

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for this and thanks for fixing up the tests too!

File: common/common.go [hunk 2]
```diff
@@ -314,12 +315,12 @@ func ExtractHost(address string) string {
...
 func ExtractPort(host string) int {
-	portStrs := strings.Split(host, ":")
-	if len(portStrs) == 1 {
+	_, port, _ := net.SplitHostPort(host)
+	if port == "" {
 		return 80
 	}
-	port, _ := strconv.Atoi(portStrs[1])
-	return port
+	portInt, _ := strconv.Atoi(port)
```
Comment (shazbert) [inline common/common.go:322]: parseUint and return uint16?

File: exchanges/asset/asset.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func Supported() Items {
 	return supportedList
 }
 
-// returns an Item to string
+// String converts an Item its string representation
```
Comment (shazbert) [inline exchanges/asset/asset.go:65]: `Item its` -> `Item to its`

File: exchanges/bitstamp/bitstamp.go [hunk 4]
```diff
@@ -654,3 +656,16 @@ func (b *Bitstamp) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange
...
+
+func filterOrderbookZeroBidPrice(ob *orderbook.Base) {
+	if len(ob.Bids) == 0 {
+		return
+	}
+
+	if ob.Bids[len(ob.Bids)-1].Price != 0 {
+		return
+	}
+
```
Comment (shazbert) [inline exchanges/bitstamp/bitstamp.go:668]:
```go
if len(ob.Bids) == 0 || ob.Bids[len(ob.Bids)-1].Price != 0 {
		return
	}
```

**Review by shazbert — CHANGES_REQUESTED** [review]
EPIC! minor changes. Also some weird test failing [here](https://ci.appveyor.com/project/thrasher-/gocryptotrader-vxsws/builds/44478175#L473)

**Review by gloriousCode — APPROVED** [review]
Thanks for making those changes! :tada:

File: common/common.go (hunk 2 above)
Comment (shazbert) [reply common/common.go:322]: Added as trello item.

**Review by shazbert — APPROVED** [review]
Have tested and checked temp folders and new linters. Looks good! Thanks for making those changes. ACK

**Pull request merged**